### PR TITLE
Add support to copy simulator device logs to output directory

### PR DESCRIPTION
**Pull request by lyndsey-ferguson**
This is useful to help debug problems that occur in the simulated app
when running UI tests on remote CI machines.

Thanks for contributing to _fastlane_! Before you submit your pull request, please make sure to check the following boxes:

- [x] Run `bundle exec rspec` from the root directory.
- [x] Run `bundle exec rubocop -a` to ensure the code style is valid
- [x] Read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] We currently don't accept new actions, please publish a plugin instead, more information in [Plugins.md](https://github.com/fastlane/fastlane/blob/master/fastlane/docs/Plugins.md)

Before submitting a pull request, we appreciate if you create an issue first to discuss the change :+1:

### CODE REVIEW

**Comment by lyndsey-ferguson** [comment]
**Note**: where should I put an automated test for this?

File: scan/lib/scan/options.rb [hunk 1]
```diff
@@ -97,6 +97,12 @@ def self.available_options
                                      env_name: "SCAN_BUILDLOG_PATH",
                                      description: "The directory were to store the raw log",
                                      default_value: "#{FastlaneCore::Helper.buildlog_path}/scan"),
+        FastlaneCore::ConfigItem.new(key: :include_simulator_logs,
+                                     env_name: "SCAN_INCLUDE_SIMULATOR_LOGS",
+                                     description: "Should the logs of the app on the Simulator be written to the output_directory",
```
Comment (KrauseFx) [inline scan/lib/scan/options.rb]: What logs are they?

File: scan/lib/scan/options.rb [hunk 2]
```diff
@@ -97,6 +97,12 @@ def self.available_options
...
+                                     type: TrueClass,
```
Comment (KrauseFx) [inline scan/lib/scan/options.rb]: Does this work like this?

File: scan/lib/scan/options.rb (hunk 1 above)
Comment (lyndsey-ferguson) [inline scan/lib/scan/options.rb]: These are the logs provided by the iOS app itself. For example, if you write `NSLog` statements in your app, these get written to the simulated device's system.log. You'll never see those log statements in your test output.

Super useful if you're trying to debug an issue that only occurs on a remote iOS CI slave and you absolutely need to know what is happening in the app.

Comment (lyndsey-ferguson) [inline scan/lib/scan/options.rb]: [This](https://github.com/soffes/SAMKeychain/issues/52) is such an issue that I solved by modifying a copy of `scan` on our Jenkins CI slave to produce the logs.

Comment (KrauseFx) [inline scan/lib/scan/options.rb]: Sounds good, do you mind clarifying the option's description to mention "the app's output (e.g. NSLog)" or similar 👍

File: scan/lib/scan/options.rb (hunk 2 above)
Comment (lyndsey-ferguson) [inline scan/lib/scan/options.rb]: Do you mean does this ConfigItem actually work? It should as I used the same code for [import_certificate](https://github.com/fastlane/fastlane/blob/master/fastlane/lib/fastlane/actions/import_certificate.rb#L35), but I have not created any tests against it and would like to know where the best place would be to write such a test (I don't see specific tests for options in `scan`)

**Comment by lyndsey-ferguson** [comment]
@KrauseFx code review comments addressed. Can you take another look? Also, is there a place that I can add a test for this, or should I add a new spec file?

**Comment by lyndsey-ferguson** [comment]
@KrauseFx bump

**Comment by lyndsey-ferguson** [comment]
@KrauseFx I improved the code to only try and copy when the simulator device log exists. I also added unit tests around this new functionality in a new file, `runner_spec.rb`.

If it looks good, can you merge? If not, let me know and I'll fix it up.

**Comment by lyndsey-ferguson** [comment]
@KrauseFx bump

**Comment by lyndsey-ferguson** [comment]
@KrauseFx bump

**Review by hjanuschka — APPROVED** [review]
nice! thx for your contribution 🚀

**Comment by lyndsey-ferguson** [comment]
Thank you @hjanuschka !

**Comment by hjanuschka** [comment]
@lyndsey-ferguson do you have any plan's on adding this to `snapshot` too? (as i do my integration tests, wherease the sim log would be awesome too )

**Comment by lyndsey-ferguson** [comment]
@hjanuschka I could definitely. We don't use it yet so my knowledge of how `snapshot` works is not as deep. Give me a bit 'o time and I'll look into it.

**Comment by lyndsey-ferguson** [comment]
Hm. It may be almost the same as `scan`; I could put it in the runner's handle results method

**Pull request merged**